# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: 05d9c3f01472
# Full Hash: 05d9c3f0147288e19525104a250b23f2de1339aa
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-09-16 10:40 +0000
# Regressor Bug: 1982261
# File Overlap Count: 1
# Overlapping Files: netwerk/cookie/CookieService.cpp
# Description:
#   Bug 1982261 - CookieServiceParent logs about invalid cookies instead of crashing, r=valentin,cookie-reviewers
# ==============================================================================

diff -r 157cf44ea0d8 -r 05d9c3f01472 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Tue Sep 16 10:23:08 2025 +0000
+++ b/netwerk/cookie/CookieService.cpp	Tue Sep 16 10:40:49 2025 +0000
@@ -1700,15 +1700,13 @@
   return mPersistentStorage;
 }
 
-bool CookieService::SetCookiesFromIPC(const nsACString& aBaseDomain,
-                                      const OriginAttributes& aAttrs,
-                                      nsIURI* aHostURI, bool aFromHttp,
-                                      bool aIsThirdParty,
-                                      const nsTArray<CookieStruct>& aCookies,
-                                      BrowsingContext* aBrowsingContext) {
+nsICookieValidation::ValidationError CookieService::SetCookiesFromIPC(
+    const nsACString& aBaseDomain, const OriginAttributes& aAttrs,
+    nsIURI* aHostURI, bool aFromHttp, bool aIsThirdParty,
+    const nsTArray<CookieStruct>& aCookies, BrowsingContext* aBrowsingContext) {
   if (!IsInitialized()) {
     // If we are probably shutting down, we can ignore this cookie.
-    return true;
+    return nsICookieValidation::eOK;
   }
 
   CookieStorage* storage = PickStorage(aAttrs);
@@ -1720,7 +1718,7 @@
     MOZ_ASSERT(validation);
 
     if (validation->Result() != nsICookieValidation::eOK) {
-      return false;
+      return validation->Result();
     }
 
     // create a new Cookie and copy attributes
@@ -1738,7 +1736,7 @@
                        aBrowsingContext);
   }
 
-  return true;
+  return nsICookieValidation::eOK;
 }
 
 void CookieService::GetCookiesFromHost(